### PR TITLE
Swift: do not assume only 1 container in tests.

### PR DESCRIPTION
**Pull request by timuralp**
Openstack Swift ContainerApiLiveTest testListWithOptions assumes that
there are no other containers and that ContainerApiLiveTest container
will _always_ be the first container.

The patch changes the behavior of the test to instead pull out the
container from the returned list.

### CODE REVIEW

File: apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java [hunk 1]
```diff
@@ -98,7 +99,13 @@ public void testListWithOptions() throws Exception {
...
          ListContainerOptions options = ListContainerOptions.Builder.marker(lexicographicallyBeforeName);
-         Container container = api.getContainerApi(regionId).list(options).get(0);
+         Container container = api.getContainerApi(regionId).list(options).firstMatch(new Predicate<Container>() {
+            @Override
+            public boolean apply(Container container) {
+               return container.getName().equals(name);
+            }
+         }).get();
+         assertNotNull(container);
          assertEquals(container.getName(), name);
```
Comment (andrewgaul) [inline apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java]: This is now always true.

File: apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java [hunk 2]
```diff
@@ -98,7 +99,13 @@ public void testListWithOptions() throws Exception {
...
       for (String regionId : regions) {
          ListContainerOptions options = ListContainerOptions.Builder.marker(lexicographicallyBeforeName);
-         Container container = api.getContainerApi(regionId).list(options).get(0);
+         Container container = api.getContainerApi(regionId).list(options).firstMatch(new Predicate<Container>() {
+            @Override
+            public boolean apply(Container container) {
+               return container.getName().equals(name);
+            }
+         }).get();
+         assertNotNull(container);
```
Comment (andrewgaul) [inline apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java]: Needed?  `get` will throw `IllegalStateException` if not present.

File: apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java (hunk 1 above)
Comment (timuralp) [inline apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java]: Done

File: apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java (hunk 2 above)
Comment (timuralp) [inline apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/ContainerApiLiveTest.java]: Done

**Comment by andrewgaul** [comment]
Pushed to master as 86af0753bf51bba42d5c18657e1f0e3dd72630ee.

**Pull request closed without merging**